src/remat.jl

```diff
@@ -458,40 +441,14 @@ function scaleinflate!(Ljj::Matrix{T}, Λj::ReMat{T,1}) where {T}
     Ljj
 end
 
-"""
-    _scaleinflateface!(A::Array, λ::Matrix, f, S)
-
-Replace the `f`'th `S×S` face of `A` by `λ'A[:,:,f]*λ + I` where `λ` is `S×S` lower triangular
-"""
-function _scaleinflateface!(A::Array, λ::Matrix, f, S)
-    @inbounds @simd for i = 1:S                   # rmul!(A[:,:,f], λ)
-        for j = 1:S
-            Aij = A[i, j, f] * λ[j, j]
-            for k = j + 1:S
-                Aij += A[i, k, f] * λ[k, j]
-            end
-            A[i, j, f] = Aij
-        end
-    end
-    for j = 1:S                  # lmul!(λ, A[:,:,f])
-        for i = 1:S
-            Aij = λ[i, i]'A[i, j, f]
-            for k = i + 1:S
-                Aij += λ[k, i]'A[k, j, f]
-            end
-            A[i, j, f] = Aij
-        end
-        A[j, j, f] += true      # inflate diagonal
-    end
-end
-
 function scaleinflate!(Ljj::UniformBlockDiagonal{T}, Λj::ReMat{T,S}) where {T,S}
-    A = Ljj.data
-    m, n, l = size(A)
-    m == n == S || throw(DimensionMismatch())
-    λ = Λj.λ.data
-    for f in 1:l
-        _scaleinflateface!(A, λ, f, S)
+    λ = Λj.λ
+    dind = diagind(S, S)
+    for f in Ljj.facevec
+        lmul!(λ', rmul!(f, λ))
+        for i in dind
+            f[i] += one(T)
```

Comment:
Adding the comment back in will help me remember what's going on when I look at this later :)
```suggestion
            f[i] += one(T) # inflate diagonal
```